### ISSUE TITLE
Cannot build for ubuntu 17.10

### ISSUE THREAD

**Issue by halaei** [redis/RedisDesktopManager#3942]
RDM Version: 0.9 (latest commit: https://github.com/uglide/RedisDesktopManager/commit/1b347261006258945361906ac6db8a97e8bc327f)
Environment (OS name and version): Ubuntu 17.10
Redis Server Version: not relevant

Steps to reproduce:

1. ./configure
Result: 
```
====================================================================
Build RDM on Ubuntu: 17
====================================================================
====================================================================
Install Qt 591
====================================================================
Hit:1 http://security.ubuntu.com/ubuntu artful-security InRelease                                                   
Hit:2 http://us.archive.ubuntu.com/ubuntu artful InRelease                                                          
Hit:3 http://us.archive.ubuntu.com/ubuntu artful-updates InRelease            
Hit:4 http://packages.microsoft.com/repos/vscode stable InRelease
Hit:5 http://us.archive.ubuntu.com/ubuntu artful-backports InRelease
Reading package lists... Done
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
====================================================================
Check deps
====================================================================
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
fatal: destination path 'src/third_party/lss' already exists and is not an empty directory.
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1000' is supported by ustar format... yes
checking whether GID '1000' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... no
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking dependency style of gcc... gcc3
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking for ranlib... ranlib
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... no
checking whether pthreads work with -Kthread... no
checking whether pthreads work with -kthread... no
checking for the pthreads library -llthread... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking a.out.h usability... yes
checking a.out.h presence... yes
checking for a.out.h... yes
checking whether g++ supports C++11 features by default... yes
checking whether C++ compiler accepts -Werror=unknown-warning-option... no
checking whether C++ compiler accepts -Wmissing-braces... yes
checking whether C++ compiler accepts -Wnon-virtual-dtor... yes
checking whether C++ compiler accepts -Woverloaded-virtual... yes
checking whether C++ compiler accepts -Wreorder... yes
checking whether C++ compiler accepts -Wsign-compare... yes
checking whether C++ compiler accepts -Wunused-local-typedefs... yes
checking whether C++ compiler accepts -Wunused-variable... yes
checking whether C++ compiler accepts -Wvla... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating breakpad.pc
config.status: creating breakpad-client.pc
config.status: creating Makefile
config.status: creating src/config.h
config.status: src/config.h is unchanged
config.status: executing depfiles commands
src/client/linux/dump_writer_common/ucontext_reader.cc: In static member function ‘static uintptr_t google_breakpad::UContextReader::GetStackPointer(const google_breakpad::ucontext*)’:
src/client/linux/dump_writer_common/ucontext_reader.cc:92:12: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RSP];
            ^~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc:92:14: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RSP];
              ^~~~~~~~~~~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc: In static member function ‘static uintptr_t google_breakpad::UContextReader::GetInstructionPointer(const google_breakpad::ucontext*)’:
src/client/linux/dump_writer_common/ucontext_reader.cc:96:12: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RIP];
            ^~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc:96:14: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RIP];
              ^~~~~~~~~~~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc: In static member function ‘static void google_breakpad::UContextReader::FillCPUContext(google_breakpad::RawContextCPU*, const google_breakpad::ucontext*, const _libc_fpstate*)’:
src/client/linux/dump_writer_common/ucontext_reader.cc:101:26: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   const greg_t* regs = uc->uc_mcontext.gregs;
                          ^~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc:101:28: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   const greg_t* regs = uc->uc_mcontext.gregs;
                            ^~~~~~~~~~~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
In file included from src/client/linux/crash_generation/crash_generation_server.cc:46:0:
./src/client/linux/handler/exception_handler.h:194:21: error: field ‘context’ has incomplete type ‘google_breakpad::ucontext’
     struct ucontext context;
                     ^~~~~~~
./src/client/linux/handler/exception_handler.h:194:12: note: forward declaration of ‘struct google_breakpad::ucontext’
     struct ucontext context;
            ^~~~~~~~
Makefile:4986: recipe for target 'src/client/linux/dump_writer_common/ucontext_reader.o' failed
make: *** [src/client/linux/dump_writer_common/ucontext_reader.o] Error 1
make: *** Waiting for unfinished jobs....
Makefile:4986: recipe for target 'src/client/linux/crash_generation/crash_generation_server.o' failed
make: *** [src/client/linux/crash_generation/crash_generation_server.o] Error 1
====================================================================
Run: source /opt/qt59/bin/qt59-env.sh && qmake && make
====================================================================

```

**Comment by uglide** [comment]
@halaei looks like you installed something in parallel using apt-get or you have broken apt-get command (broken lock).

**Comment by halaei** [comment]
I don't think so!
What is this error?
```
src/client/linux/dump_writer_common/ucontext_reader.cc:92:12: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RSP];
            ^~
```

**Comment by uglide** [comment]
First error in your output is:
```
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
```

**Comment by halaei** [comment]
That was passed. I tried it again:
```
====================================================================
Build RDM on Ubuntu: 17
====================================================================
====================================================================
Install Qt 591
====================================================================
[sudo] password for hamid: 
Ign:1 http://dl.google.com/linux/chrome/deb stable InRelease                                                                                                                                               
Get:2 http://security.ubuntu.com/ubuntu artful-security InRelease [78.6 kB]                                     
Hit:3 http://us.archive.ubuntu.com/ubuntu artful InRelease                                                                  
Hit:4 http://packages.microsoft.com/repos/vscode stable InRelease                                                           
Hit:5 http://archive.canonical.com/ubuntu artful InRelease                                                                  
Get:6 http://dl.google.com/linux/chrome/deb stable Release [1,189 B]                                       
Get:7 http://us.archive.ubuntu.com/ubuntu artful-updates InRelease [78.6 kB]                                       
Get:8 http://dl.google.com/linux/chrome/deb stable Release.gpg [819 B]                     
Get:9 http://dl.google.com/linux/chrome/deb stable/main amd64 Packages [1,388 B]
Get:10 http://us.archive.ubuntu.com/ubuntu artful-backports InRelease [72.2 kB]                                                                                                                            
Fetched 233 kB in 10s (21.6 kB/s)                                                                                                                                                                          
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package qt59base
E: Unable to locate package qt59imageformats
E: Unable to locate package qt59tools
E: Unable to locate package qt59declarative
E: Unable to locate package qt59quickcontrols
E: Unable to locate package qt59charts-no-lgpl
E: Unable to locate package qt59svg
====================================================================
Check deps
====================================================================
Reading package lists... Done
Building dependency tree       
Reading state information... Done
automake is already the newest version (1:1.15-6ubuntu1).
cmake is already the newest version (3.9.1-1).
g++ is already the newest version (4:7.2.0-1ubuntu1).
libgl1-mesa-dev is already the newest version (17.2.2-0ubuntu1).
libtool is already the newest version (2.4.6-2).
libssh2-1-dev is already the newest version (1.8.0-1).
libssl-dev is already the newest version (1.0.2g-1ubuntu13.3).
The following packages were automatically installed and are no longer required:
  linux-headers-4.13.0-16 linux-headers-4.13.0-16-generic linux-image-4.13.0-16-generic linux-image-extra-4.13.0-16-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
fatal: destination path 'src/third_party/lss' already exists and is not an empty directory.
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1000' is supported by ustar format... yes
checking whether GID '1000' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... no
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking dependency style of gcc... gcc3
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking for ranlib... ranlib
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... no
checking whether pthreads work with -Kthread... no
checking whether pthreads work with -kthread... no
checking for the pthreads library -llthread... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking a.out.h usability... yes
checking a.out.h presence... yes
checking for a.out.h... yes
checking whether g++ supports C++11 features by default... yes
checking whether C++ compiler accepts -Werror=unknown-warning-option... no
checking whether C++ compiler accepts -Wmissing-braces... yes
checking whether C++ compiler accepts -Wnon-virtual-dtor... yes
checking whether C++ compiler accepts -Woverloaded-virtual... yes
checking whether C++ compiler accepts -Wreorder... yes
checking whether C++ compiler accepts -Wsign-compare... yes
checking whether C++ compiler accepts -Wunused-local-typedefs... yes
checking whether C++ compiler accepts -Wunused-variable... yes
checking whether C++ compiler accepts -Wvla... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating breakpad.pc
config.status: creating breakpad-client.pc
config.status: creating Makefile
config.status: creating src/config.h
config.status: src/config.h is unchanged
config.status: executing depfiles commands
src/client/linux/dump_writer_common/ucontext_reader.cc: In static member function ‘static uintptr_t google_breakpad::UContextReader::GetStackPointer(const google_breakpad::ucontext*)’:
src/client/linux/dump_writer_common/ucontext_reader.cc:92:12: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RSP];
            ^~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc:92:14: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RSP];
              ^~~~~~~~~~~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc: In static member function ‘static uintptr_t google_breakpad::UContextReader::GetInstructionPointer(const google_breakpad::ucontext*)’:
src/client/linux/dump_writer_common/ucontext_reader.cc:96:12: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RIP];
            ^~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc:96:14: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   return uc->uc_mcontext.gregs[REG_RIP];
              ^~~~~~~~~~~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc: In static member function ‘static void google_breakpad::UContextReader::FillCPUContext(google_breakpad::RawContextCPU*, const google_breakpad::ucontext*, const _libc_fpstate*)’:
src/client/linux/dump_writer_common/ucontext_reader.cc:101:26: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   const greg_t* regs = uc->uc_mcontext.gregs;
                          ^~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
src/client/linux/dump_writer_common/ucontext_reader.cc:101:28: error: invalid use of incomplete type ‘const struct google_breakpad::ucontext’
   const greg_t* regs = uc->uc_mcontext.gregs;
                            ^~~~~~~~~~~
In file included from src/client/linux/dump_writer_common/ucontext_reader.cc:30:0:
./src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of ‘struct google_breakpad::ucontext’
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
In file included from src/client/linux/crash_generation/crash_generation_server.cc:46:0:
./src/client/linux/handler/exception_handler.h:194:21: error: field ‘context’ has incomplete type ‘google_breakpad::ucontext’
     struct ucontext context;
                     ^~~~~~~
./src/client/linux/handler/exception_handler.h:194:12: note: forward declaration of ‘struct google_breakpad::ucontext’
     struct ucontext context;
            ^~~~~~~~
Makefile:4986: recipe for target 'src/client/linux/dump_writer_common/ucontext_reader.o' failed
make: *** [src/client/linux/dump_writer_common/ucontext_reader.o] Error 1
make: *** Waiting for unfinished jobs....
Makefile:4986: recipe for target 'src/client/linux/crash_generation/crash_generation_server.o' failed
make: *** [src/client/linux/crash_generation/crash_generation_server.o] Error 1
====================================================================
Run: source /opt/qt59/bin/qt59-env.sh && qmake && make
====================================================================

```

**Comment by halaei** [comment]
What fixes these errors:
```
E: Unable to locate package qt59base
E: Unable to locate package qt59imageformats
E: Unable to locate package qt59tools
E: Unable to locate package qt59declarative
E: Unable to locate package qt59quickcontrols
E: Unable to locate package qt59charts-no-lgpl
E: Unable to locate package qt59svg
```
Should I install every libqt5* package?

**Comment by uglide** [comment]
Read configure script and copy-paste Qt installing from there

**Comment by halaei** [comment]
@uglide It seems there is a compatibility issue with new glibc. Here is a related issue in another project with a merged PR:
https://github.com/voidlinux/void-packages/issues/7599
https://github.com/voidlinux/void-packages/pull/7605


To fix that, I find I just need to
```
cd 3rdparty/gbreakpad/ && git checkout master
```

**Comment by uglide** [comment]
@halaei will be fixed in #3947
